packages/odata-v4/src/de-serializers/temporal-de-serializers.ts

```diff
@@ -0,0 +1,126 @@
+/* eslint-disable valid-jsdoc */
+import { Temporal } from '@js-temporal/polyfill';
+import { DeSerializer } from '@sap-cloud-sdk/odata-common/internal';
+/**
+ * Temporal (de-)serializers for Odata-v4.
+ * @internal
+ */
+export const temporalDeSerializers = {
+  'Edm.Date': {
+    deserialize: deserializeDateToTemporal,
+    serialize: serializePlainDateToDate,
+    serializeToUri: (value, serialize) => serialize(value)
+  } as DeSerializer<Temporal.PlainDate>,
+  'Edm.DateTimeOffset': {
+    deserialize: deserializeDateTimeOffsetToTemporal,
+    serialize: serializeZonedDateTimeToDateTimeOffset,
+    serializeToUri: (value, serialize) => serialize(value)
+  } as DeSerializer<Temporal.ZonedDateTime>,
+  'Edm.Duration': {
+    deserialize: deserializeDurationToTemporal,
+    serialize: serializeDurationToDuration,
+    serializeToUri: (value, serialize) => `duration'${serialize(value)}'`
+  } as DeSerializer<Temporal.Duration>,
+  'Edm.TimeOfDay': {
+    deserialize: deserializeTimeToTemporal,
+    serialize: serializePlainTimeToTime,
+    serializeToUri: (value, serialize) => serialize(value)
+  } as DeSerializer<Temporal.PlainTime>
+};
+
+/**
+ * @internal
+ */
+export function serializePlainTimeToTime(value: Temporal.PlainTime): string {
+  return value.toString();
+}
+
+/**
+ * @internal
+ */
+export function deserializeDateToTemporal(date: string): Temporal.PlainDate {
+  const dateFormat = /\d{4}-(0[1-9]|1[012])-(0[1-9]|[12][0-9]|3[01])/;
```

Comment:
could you leave a comment that this is the regex for YYYY-MM-DD. It is nice that you enforced the numbers on month either 0X or 1Y and day but you did not do this on year where you also have some conditions. So I would perhaps loosen the regex to just `\d{2}". Also I wonder if this could be delegated to the `from` method. What would happen if a wrong string is parse?

Reply:
for the `deserializeDateTimeOffsetToTemporal` you did it this way